lib/game/parallel.go

```diff
@@ -0,0 +1,251 @@
+package game
+
+import (
+	"sync"
+)
+
+type NeighborData struct {
+	Generation uint32
+	Data       []uint8
+}
+
+type NeighborConnection struct {
+	SendCh    chan NeighborData
+	ReceiveCh chan NeighborData
+	Data      NeighborData
+}
+
+func NewNeighborConnection() *NeighborConnection {
+	return &NeighborConnection{
+		SendCh: make(chan NeighborData, 1),
+	}
+}
+
+type ParallelUniverse struct {
+	*Universe
+
+	TopNeighbor    *NeighborConnection
+	BottomNeighbor *NeighborConnection
+	LeftNeighbor   *NeighborConnection
+	RightNeighbor  *NeighborConnection
+}
+
+func NewParallelUniverse(height, width uint32) *ParallelUniverse {
+	p := &ParallelUniverse{
+		Universe: NewUniverse(height, width),
+	}
+	p.Rules = p.ParallelRules
+	return p
+}
+
+func (p *ParallelUniverse) ParallelRules(cell uint8, row, column uint32) uint8 {
+	return RuleB3S23(cell, p.ParallelNeighbors(row, column))
+}
+
+// ParallelNeighbors returns the number of alive neighbors for a given cell.
+// It uses the Moore neighborhood, which includes the eight cells surrounding
+// the given cell, but also extended to adjoining neighbor Universes.
+func (p *ParallelUniverse) ParallelNeighbors(row, column uint32) uint8 {
+	count := uint8(0)
+	r, c := int32(row), int32(column)
+	for _, neighborRow := range []int32{r - 1, r, r + 1} {
+		for _, neighborColumn := range []int32{c - 1, c, c + 1} {
+			if neighborRow == r && neighborColumn == c {
+				// Skip checking the cell itself
+				continue
+			}
+
+			switch {
+			case neighborRow < 0 && neighborColumn < 0:
+				// check the universe above and to the left?
```

Comment:
we either have to also connect diagonal universes or expose a function to check for these cases

Reply:
It is a check for only 1 pixel which requires connecting to a whole `Universe` :crying_cat_face: 

Not sure what to do about it at the moment, hence why the comment. Perhaps ignore for now?